src/modules/reservation.js

```diff
@@ -0,0 +1,126 @@
+
+    const h2 = document.createElement('h2');
+    h2.textContent = show.show.name;
+
+    const summary = document.createElement('p');
+    summary.innerHTML = show.show.summary;
+
+    const reservationForm = document.createElement('form');
+    reservationForm.className = 'reservation-form';
+
+    const nameInput = document.createElement('input');
+    nameInput.type = 'text';
+    nameInput.placeholder = 'Name';
+
+    const startDateInput = document.createElement('input');
+    startDateInput.type = 'date';
+    startDateInput.placeholder = 'Start Date';
+
+    const endDateInput = document.createElement('input');
+    endDateInput.type = 'date';
+    endDateInput.placeholder = 'End Date';
+
+    const reserveButton = document.createElement('button');
+    reserveButton.textContent = 'Reserve';
+    const reservationsContainer = document.createElement('div');
+    reservationsContainer.className = 'reservations-container';
+
+    const totalReservations = document.createElement('p');
+    totalReservations.className = 'total-reservations';
+    // eslint-disable-next-line no-inner-declarations
+    async function fetchTotalReservations(itemId) {
+      try {
+        const apiKey = 'fDqwhSI3IFsJVLYNBHB6';
+        const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${apiKey}/reservations?item_id=${itemId}`);
+
+        if (!response.ok) {
+          throw new Error('Failed to fetch reservations');
+        }
+
+        const reservations = await response.json();
+        return reservations.length;
+      } catch (error) {
+        console.error('Error fetching total reservations:', error);
+        return 0;
+      }
+    }
+    // eslint-disable-next-line no-inner-declarations
+    async function displayTotalReservations(itemId) {
+      const total = await fetchTotalReservations(itemId);
+      totalReservations.textContent = `Total Reservations: ${total}`;
+    }
+
+    reservationForm.addEventListener('submit', async (e) => {
+      e.preventDefault();
+
+      const reservationData = {
+        item_id: show.show.id,
+        username: nameInput.value,
+        date_start: startDateInput.value,
+        date_end: endDateInput.value,
+      };
+
+      const apiKey = 'fDqwhSI3IFsJVLYNBHB6';
+      const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${apiKey}/reservations`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(reservationData),
+      });
+      /* eslint-disable no-alert */
+      if (response.ok) {
+        nameInput.value = '';
+        startDateInput.value = '';
+        endDateInput.value = '';
+        displayReservations(show.show.id, reservationsContainer);
+        displayTotalReservations(show.show.id);
+      } else {
+        alert('Reservation failed. Please try again.');
+      }
+    });
+    /* eslint-enable no-alert */
```

Comment:
Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.